### PR TITLE
Add JSON parsing to File resolver

### PR DESCRIPTION
**Pull request by bbatsche**

### CODE REVIEW

File: src/Resolver/File.php [hunk 1]
```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
             $path = realpath($this->getIterator()->getRootDefinition()->getBasepath() . \DIRECTORY_SEPARATOR . $path);
         }
 
-        return file_get_contents($path);
+        $content = file_get_contents($path);
+
+        if (pathinfo($path, PATHINFO_EXTENSION) == 'json') {
```
Comment (tjwiebell) [inline src/Resolver/File.php]: We can't assume we need to parse based on file extension, should have this logic:

1. `$parse === 'auto' && pathinfo($path, PATHINFO_EXTENSION) == 'json'` (auto parse)
2. `$parse === 'json'` (explicit parse)

I think the first gate covers it, but `$parse === 'text'` should return raw content, to allow raw json to still be returned if needed.

File: src/Resolver/File.php [hunk 2]
```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
...
+            $content = json_decode($content, true);
+
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                throw new \RuntimeException('Failed to parse ' . basename($path) . ': ' . json_last_error_msg());
```
Comment (tjwiebell) [inline src/Resolver/File.php]: Just a comment, `upward-js` doesn't have json validation. Should we suggest it over on magento-research/pwa-studio#812?

File: src/Resolver/File.php (hunk 1 above)
Comment (bbatsche) [reply src/Resolver/File.php]: Whoops! Good catch

File: src/Resolver/File.php (hunk 2 above)
Comment (bbatsche) [reply src/Resolver/File.php]: Are you sure `JSON.parse()` doesn't already throw an exception/error for invalid syntax? PHP's the difficult on in this case.

Comment (tjwiebell) [reply src/Resolver/File.php]: Ah, yes; JS does throw a SyntaxError.

File: src/Resolver/File.php [hunk 3]
```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
             $path = realpath($this->getIterator()->getRootDefinition()->getBasepath() . \DIRECTORY_SEPARATOR . $path);
         }
 
-        return file_get_contents($path);
+        $content = file_get_contents($path);
+
+        if ($parse == 'auto' && pathinfo($path, PATHINFO_EXTENSION) == 'json') {
```
Comment (tjwiebell) [inline src/Resolver/File.php]: Do we need type coercion here? Also missed explicit json parsing (edge case where json might be in a `json.txt` file maybe?).
```suggestion
        if (($parse === 'auto' && pathinfo($path, PATHINFO_EXTENSION) === 'json') || $parse === 'json') {
```

Comment (bbatsche) [reply src/Resolver/File.php]: I think that's actually an ambiguity in the upward spec: https://github.com/magento-research/pwa-studio/tree/release/2.0/packages/upward-spec#fileresolver-configuration-options. I interpreted that to mean that the acceptable values for `parse` are either `"text"` or `"auto"`, but the paragraph after implies that other values (including `"json"`) *may* be acceptable.

Comment (tjwiebell) [reply src/Resolver/File.php]: Confirmed with the architect, explicit json parsing is an expected feature. I'll review the `upward-spec` docs and suggest an edit if there's a change that would make that more clear.

**Pull request merged**